DOCUMENTATION.md

```diff
@@ -0,0 +1,101 @@
+-   `{{% docker-version %}}` - Outputs the value of `Params.releases.docker` as defined in the site configuration to output the consistent current Docker release.
+-   `{{% now %}}` - Outputs the current Unix timestamp to allow for up to date timestamps in code examples.
+
+## Running Documentation
+
+As noted in the prerequisites section, if you want to run the documentation locally with `hugo server`, you need to have the Victor theme in the same directory as the M3 codebase, as `hugo server` runs in Hugo's "development" mode (and matches _site/config/development/config.toml_).
+
+This does mean that as you make changes to the theme or documentation, usually, it refreshes automatically in the browser preview. Sometimes Hugo doesn't refresh included files, so you may need to restart the server process.
+
+## Testing Documentation
+
+The M3 documentation has a number of tests that need to pass before a pull request is merged to master and deployed. These include:
+
+-   Link checking with htmltest
+
+To run them you need Docker installed and running, and at the top level of the M3 project, run `make clean install-vendor-m3 docs-test`.
+
+## Building Documentation
+
+There are a couple of different ways to build the documentation depending what you want to do.
+
+If you want to build only the most up-to-date version of the docs, you can use `hugo` to build.
+
+If you want to build all versions of the docs using Hugo run in Docker (this is what CI does to test the documentation). From the top level of the project, run `make docs-build`.
+
+If you want to build all versions of the docs with a system-installed version of Hugo (this is what Netlify does to build and serve the documentation). From the top level of the project, run `make site-build`.
+
+## Creating a New Documentation Version
+
+M3 releases versions with some slight changes to documentation for each one which users can access from the drop down menu under the right hand navigation.
+
+Archiving a version of the documentation is a slightly complex process as Hugo doesn't natively support versioning.
+
+1.  Add the new version and path to _site/config/production/config.toml_ in the format:
+
+    ```toml
+    [[params.versions]]
+      version = "{versionNumber}"
+      url = "/v{versionNumber}/docs"
+    ```
+2.  Add the `/v{versionNumber}` path before each `{{ .RelPermalink}}` in the two anchor tags in _site/layouts/partials/navbar.html_
+3.  Make sure the Victor theme is at least version 0.2.11
+4.  Create and tag the commits with the name "docs/v{versionNumber}"
+5.  Switch to a new branch that will add the new version to the master branch
+6.  Add the new version to _site/config/production/config.toml_
+7.  Also add the new version to _site/config/production/config.toml_ in every pre-existing docs version tag
+8.  Switch back to the branch from step 5
+9.  Test everything locally with `make site-build`
+10. Push the new tag to the remote repository
```

Comment:
Do we want to link to the contribution guide as well? Or add a section on the review process once the contribution / docs PR has been pushed?